src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -143,11 +143,7 @@ export class CalciteRadioButton implements LabelableComponent, CheckableFormComp
   //--------------------------------------------------------------------------
 
   selectItem = (items: HTMLCalciteRadioButtonElement[], selectedIndex: number): void => {
-    items.forEach((item, index) => {
-      const selected = index === selectedIndex;
-      item.checked = selected;
-      item.focused = selected;
-    });
+    items[selectedIndex].click();
```

Comment:
Could you make sure that this works properly when `checked` is set programmatically in a group?

Reply:
This line won't fire when checked is programmatically set, because this only gets called on keyboard interactions.  Plus, there are already some tests that cover this: https://github.com/Esri/calcite-components/blob/9d4c9667b9f4a0ea396e286de5ade597c36fccf1/src/components/calcite-radio-button/calcite-radio-button.e2e.ts#L196-L231

Reply:
Excellent, thanks for confirming.